### PR TITLE
Added Test Publisher

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description

- Implemented Test Publisher to store result of the test to S3. 
- Path to test-results `bundlebuild/${opensearch_version}/${build_id}/${platform}/${architecture}/test-results/test_run_id/test_type/component`. 
- Added `platform` as a build parameter.
- Fixed run command for Integ Test to execute without S3 dependencies

Ref: https://github.com/opensearch-project/opensearch-build/issues/207 for the discussion.
 
### Issues Resolved
https://github.com/opensearch-project/opensearch-build/issues/341

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -9,6 +9,7 @@
 set -e
 
 DIR="$(dirname "$0")"
+mkdir -p 'test-results'
```
Comment (dblock) [inline test.sh:12]: Let's do this in code before we write results like for all the other paths (we create directories for builds and dist).

**Review by dblock — CHANGES_REQUESTED** [review]

File: jenkins/test/testsuite/Jenkinsfile [hunk 2]
```diff
@@ -34,6 +34,11 @@ pipeline {
                                             name: 'test_run_id',
```
Comment (gaiksaya) [inline jenkins/test/testsuite/Jenkinsfile:34]: AFAIR, test_run_id is the jenkins `build_number`. Should that be set as default value for `test_run_id` param? That is not something a user will or should set

Comment (owaiskazi19) [reply jenkins/test/testsuite/Jenkinsfile:34]: Coming from the code: https://github.com/opensearch-project/opensearch-build/blob/main/src/run_integ_test.py#L33. `test_run_id` is passed like this. It's safe to remove it from the param.

Comment (owaiskazi19) [reply jenkins/test/testsuite/Jenkinsfile]: Pushed the change.

Comment (gaiksaya) [reply jenkins/test/testsuite/Jenkinsfile]: It is being retrieved from TestArgs() https://github.com/opensearch-project/opensearch-build/blob/9bc0a41493d99b2c03f5193ca5d0437b6609907f/src/test_workflow/test_args.py#L34

Comment (gaiksaya) [reply jenkins/test/testsuite/Jenkinsfile]: I dont think it should be removed. How will we know which test results are from which workflow run. Or is that information not relevant?

@peternied @dblock thoughts? Also don't know why we added `uuid.uuid4().hex` for test_run_id.

Comment (owaiskazi19) [reply jenkins/test/testsuite/Jenkinsfile]: Are you suggesting a path like `bundlebuild/${opensearch_version}/${build_id}/${platform}/${architecture}/test_run_id/test-results/test_type/component`?

Comment (gaiksaya) [reply jenkins/test/testsuite/Jenkinsfile]: Nope that is take care of in test_recorder. Ask is just set the `$BUILD_NUMBER` as default value of `test_run_id` in jenkins parameters.

Comment (owaiskazi19) [reply jenkins/test/testsuite/Jenkinsfile]: I have removed `test_run_id` from Jenkinsfile as test_run_id is already being set in the code mentioned above.

**Review by dblock — APPROVED** [review]

File: jenkins/test/testsuite/Jenkinsfile (hunk 2 above)
Comment (owaiskazi19) [reply jenkins/test/testsuite/Jenkinsfile]: As discussed offline with @gaiksaya. I'll change the path to `bundlebuild/${opensearch_version}/${build_id}/${platform}/${architecture}/test_run_id/test-results/test_type/component` to record BUILD_NUM as the test_run_id

File: jenkins/test/testsuite/Jenkinsfile [hunk 3]
```diff
@@ -59,13 +60,28 @@ pipeline {
                 }
             }
             steps {
-                sh "./test.sh ${JOB_NAME} --s3-bucket ${ARTIFACT_BUCKET_NAME} --opensearch-version ${opensearch_version} --build-id ${build_id} --architecture ${architecture} --test-run-id ${test_run_id}"
+                script {
+                    basePath = "https://ci.opensearch.org/ci/dbc/bundle-build/${opensearch_version}/${build_id}/${platform}/${architecture}"
```
Comment (peternied) [inline jenkins/test/testsuite/Jenkinsfile:64]: Can we make this a parameter to the job? We just had a build break yesterday night due to building these paths inconsistently

Comment (owaiskazi19) [reply jenkins/test/testsuite/Jenkinsfile:64]: `https://ci.opensearch.org/ci/dbc/bundle-build/`should be a parameter to the job?

Comment (peternied) [reply jenkins/test/testsuite/Jenkinsfile:64]: Even more - ` basePath = "https://ci.opensearch.org/ci/dbc/bundle-build/${opensearch_version}/${build_id}/${platform}/${architecture}"` could be passed in a parameter, only the architecture is used to determine the agent node type all the rest of the values are passed around to no effect.

...lets come back to this after you merge to build momentum

**Review by peternied — APPROVED** [review]

**Review by gaiksaya — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**